### PR TITLE
Record container id and type in core job metrics

### LINKED ISSUE #16588: Show container resolution and actual used container in job-info page
xref: https://github.com/galaxyproject/galaxy/issues/13037#issuecomment-991285167

```
One major problem is that only the dependency resolution is shown on that page currently, not the container resolution. Additionally, I am pretty sure that you're seeing the dependency as it resolves presently, not the dependency that was necessarily used when the job ran.
```

And I think @mvdbeek suggested running it as a metric plugin.

### PR DESCRIPTION
**Pull request by mvdbeek**
Closes https://github.com/galaxyproject/galaxy/issues/16588. Not sure if we want to do this versus a separate but always loaded plugin vs a regular relationship in the database.

Seems unlikely that we'd do anything beyond reporting, so maybe this is fine for now ?

Screenshot 2024-07-20 at 22 48 00

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/job_metrics/instrumenters/core.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
+import json
 import logging
+import os
 import time
 from typing import (
     Any,
     Dict,
     List,
+    Optional,
```
Comment (jmchilton) [inline lib/galaxy/job_metrics/instrumenters/core.py]: Just drop this.

**Comment by jmchilton** [comment]
This has been... honestly very high... on my priority list for like 9 years at this point I think 😅.

**Comment by mvdbeek** [comment]
This has changed a little, still ok @jmchilton ?

**Comment by jmchilton** [comment]
```
FAILED test/integration/test_kubernetes_runner.py::TestKubernetesIntegration::test_explicit - AttributeError: 'TestKubernetesIntegration' object has no attribute 'container_type'
```

is this related?

**Comment by mvdbeek** [comment]
Yep, thanks!

**Comment by nsoranzo** [comment]
Sorry for the late review, I think you need to add the `container_type` class attribute also to https://github.com/galaxyproject/galaxy/blob/dev/test/integration/test_coexecution.py#L242 ? Tests pass because this is skipped in GitHub workflows.

**Pull request merged**